lib/awful/placement.lua

```diff
@@ -493,7 +496,7 @@ function placement.no_overlap(c)
     new.width = geometry.width
     new.height = geometry.height
 
-    return c:geometry({ x = new.x, y = new.y })
+    return c:geometry({ x = new.x + c.border_width, y = new.y + c.border_width })
 end
```

Comment:
The problem you were seeing is that `placement.no_offscreen()` would move a client by its border width, right?
I think your patch only hides the problem, but does not fix it. The actual bug is in `area_common` which does:
```
geometry.x = geometry.x - border
geometry.y = geometry.y - border
geometry.width = geometry.width + 2 * border
geometry.height = geometry.height + 2 * border
```
Adding the border to the width is correct, but subtracting it from (x,y) is incorrect. The position is outside of the border, not inside of it.

Thus, because of the above subtraction, you get the wrong values here.

Put different: `c:geometry{ x = c:geometry().x }` does not modify the position of a client, but adding the border width does move it. So if your patch is needed, then something is messing up the position (this something is `area_common()`).